src/middleware/trailing-slash/index.test.ts

```diff
@@ -0,0 +1,43 @@
+import { Hono } from '../../hono'
+import { trimTrailingSlash, appendTrailingSlash } from '.'
+
+describe('Resolve trailing slash', () => {
+  let app: Hono
+
+  it('Trim', async () => {
+    app = new Hono({ strict: true })
+
+    app.use('*', trimTrailingSlash())
+
+    app.get('/the/example/endpoint/without/trailing/slash', async (c) => {
+      return c.text('301 moved permanently')
+    })
+
+    const resp = await app.request('/the/example/endpoint/without/trailing/slash/')
+    expect(resp).not.toBeNull()
+    expect(resp.status).toBe(301)
+  })
```

Comment:
Can you add a test when the request path is `/the/example/endpoint/without/trailing/slash` and `/`?

Reply:
Per the reason provided, currently, I'm adding a test for `/` and awaiting for further discussions.